### PR TITLE
Make image more flexible

### PR DESCRIPTION
**Pull request by hhorak**
Scripts run during assemble and run phase are moved to different directory, so it's possible to rewrite that behaviour.
Include some simple logic for SSL certificates, so users can provide their own PEM files.

### CODE REVIEW

**Comment by hhorak** [comment]
This is a similar feature that has been requested for PHP image in https://bugzilla.redhat.com/show_bug.cgi?id=1478556. I'd like to first make this available in pure httpd image, and then use the same logic for PHP.
@remicollet @uhliarik @notroj WDYT?

**Comment by remicollet** [comment]
@hhorak thanks for taking care of this.

Looks good to me.

Only a question: do the server name in the container may raise some issue with the name used in the certificate ?

File: 2.4/root/usr/share/container-scripts/httpd/common.sh [hunk 1]
```diff
@@ -90,3 +90,41 @@ generate_container_user() {
   export NSS_WRAPPER_GROUP=/etc/group
 }
 
+# Copy config files from application to the location where httd expects them
```
Comment (mavit) [inline 2.4/root/usr/share/container-scripts/httpd/common.sh]: Typo here, "httd".

**Comment by notroj** [comment]
This is great @hhorak - thank you!

It actually makes more sense to make the key files optional, since with current mod_ssl you can put the server certificate and private key (plus optionally other stuff like any intermediate CA certs) in a single file passed to SSLCertificateFile.

Is this ready to merge? I can make this change after merging the PR.

File: 2.4/root/usr/share/container-scripts/httpd/common.sh [hunk 2]
```diff
@@ -90,3 +90,41 @@ generate_container_user() {
...
+  local dir=${1:-.}
+  if [ -d ${dir}/httpd-ssl/private ] && [ -d ${dir}/httpd-ssl/certs ]; then
+    echo "---> Looking for SSL certs for httpd..."
+    cp -r ${dir}/httpd-ssl ${HTTPD_APP_ROOT}
+    local ssl_cert="$(ls -A ${HTTPD_APP_ROOT}/httpd-ssl/certs/*.pem | head -n 1)"
+    local ssl_private="$(ls -A ${HTTPD_APP_ROOT}/httpd-ssl/private/*.pem | head -n 1)"
+    if [ -f "${ssl_cert}" ] && [ -f "${ssl_private}" ]; then
+      echo "---> Setting SSL certs for httpd..."
+      sed -i -e "s|^SSLCertificateFile .*$|SSLCertificateFile ${ssl_cert}|" ${HTTPD_MAIN_CONF_D_PATH}/ssl.conf
+      sed -i -e "s|^SSLCertificateKeyFile .*$|SSLCertificateKeyFile ${ssl_private}|" ${HTTPD_MAIN_CONF_D_PATH}/ssl.conf
```
Comment (notroj) [inline 2.4/root/usr/share/container-scripts/httpd/common.sh]: So keeping this branch but adding an
```
elif [ -f "${ssl_cert}" ]
  # do sed for SSLCertificateFile
  sed -i /SSLCertificateKeyFile .*/d  ${HTTPD_MAIN_CONF_D_PATH}/ssl.conf
fi
```
is probably right, i.e. remove SSLCertificateKeyFile if *only* a cert file is present.

Comment (hhorak) [reply 2.4/root/usr/share/container-scripts/httpd/common.sh]: Thanks for this suggestion, I'll change it this way.

**Comment by hhorak** [comment]
> Only a question: do the server name in the container may raise some issue with the name used in the certificate ?

@remicollet Good question, I don't know httpd configuration that well. Do you think we should make the ServerName also configurable easily?

**Comment by hhorak** [comment]
It looks like there are no objections. @remicollet, I'm still wondering what you meant by the ServerName, so feel free to propose a change if you feel it's needed.

**Pull request merged**